lib/sycamore/sycamore/writer.py

```diff
@@ -800,10 +822,17 @@ def json(
 
         self._maybe_execute(node, True)
 
-    def _maybe_execute(self, node: Node, execute: bool) -> Optional[DocSet]:
+    def _maybe_execute(
+        self, node: Node, execute: bool, client: Optional[OpenSearchWriterClient] = None
+    ) -> Optional[DocSet]:
         ds = DocSet(self.context, node)
         if not execute:
             return ds
 
         ds.execute()
+        if client is not None:
```

Comment:
fix this to just call client.reliabliity_assertor(...) and do the raise in the client base class.